Druid/DRRegisterCollapse.class.st

```diff
@@ -0,0 +1,13 @@
+Class {
+	#name : #DRRegisterCollapse,
+	#superclass : #DROptimisation,
+	#category : #'Druid-BytecodeGeneration'
+}
+
+{ #category : #accessing }
+DRRegisterCollapse >> applyTo: cfg [
+
+	cfg instructionsDo: [ :i | 
+		i isMandatoryInstruction ifFalse: [ 
+			i basicBlock removeInstruction: i ] ]
+]
```

Comment:
This is here because the transformation from register-based model (Druid) to a stack-based (Bytecode).

Now, many instructions are removed from CFG to be iterated in different way.

We need to look a ~better~ real strategy for this. Check https://www.sciencedirect.com/science/article/pii/S1571066111001538?ref=pdf_download&fr=RR-2&rr=7598ca93ec5e7761